app/controllers/application_controller.rb

```diff
@@ -22,21 +22,22 @@ def after_sign_out_path_for(_resource)
   # for production or having setup secrets.
   # If they are not met, render a page with status 500
   def check_requirements
-    fix_secrets, fix_ssl = fixes
-    return unless fix_secrets || fix_ssl
-
-    text = "Please review the following configurations"
-    text += "<ul>"
-    text += "<li>ssl</li>" if fix_ssl
-    text += "<li>secrets</li>" if fix_secrets
-    text += "</ul>"
-    render text: text, status: 500
+    # select any model to check database connection
+    # if database is not configured it will raise an exception
+    User.first
+    fix= fixes
+    return unless fix.has_value?(true)
+    redirect_to "/errors/500?fix_ssl=#{fix[:ssl]}&fix_secret_key_base=#{fix[:secret_key_base]}&fix_secret_machine_fqdn=#{fix[:secret_machine_fqdn]}&fix_secret_encryption_private_key_path=#{fix[:secret_encryption_private_key_path]}&fix_secret_portus_password=#{fix[:secret_portus_password]}", status: 500
   end
 
   def fixes
-    fix_secrets = true if Rails.application.secrets.secret_key_base == "CHANGE_ME"
-    fix_ssl     = true if Rails.env.production? && !request.ssl?
-    [fix_secrets, fix_ssl]
+    fix = {}
```

Comment:
use `tap`, as it's done here: https://github.com/SUSE/Portus/blob/master/app/models/jwt_token.rb#L12